src/context.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright Copyright 2024, Horizen Labs, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use alloc::{collections::BTreeMap, string::String};
+use risc0_zkp::{
+    core::hash::{
+        blake2b::Blake2bCpuHashSuite, poseidon2::Poseidon2HashSuite, sha::Sha256HashSuite,
+        HashSuite,
+    },
+    field::baby_bear::BabyBear,
+};
+
+use crate::{
+    circuit::{self, CircuitCoreDef},
+    receipt::succinct::SuccinctReceiptVerifierParameters,
+    segment::SegmentReceiptVerifierParameters,
+};
+
+/// Context available to the verification process.
```

Comment:
Maybe add here or in the readme some rationale about the choice of the kind of verifiers we support